File: ios/Classes/document/Document.swift
class Document {
    let id: String
    let renderer: CGPDFDocument
    var pages: [CGPDFPage?]
    
    init(id: String, renderer: CGPDFDocument) {
        self.id = id
        self.renderer = renderer
        self.pages = Array<CGPDFPage?>(repeating: nil, count: renderer.numberOfPages)
    }
    
    var pagesCount: Int {
        get {
            return renderer.numberOfPages
        }
    }
    
    var infoMap: [String: Any] {
        get {
            return [
                "id": id,
                "pagesCount": Int32(pagesCount)
            ]
        }
    }
    
    /**
     * Open page by page number (not index!)
     */
    public func openPage(pageNumber: Int) -> CGPDFPage? {
        return renderer.page(at: pageNumber)
    }
}

File: packages/pdfx/android/src/main/kotlin/io/scer/pdfx/document/Document.kt
package io.scer.pdfx.document

import android.annotation.TargetApi
import android.graphics.pdf.PdfRenderer
import android.os.Build
import android.os.ParcelFileDescriptor

@TargetApi(Build.VERSION_CODES.LOLLIPOP)
class Document (
    val id: String,
    private val documentRenderer: PdfRenderer,
    private val fileDescriptor: ParcelFileDescriptor
) {
    val pagesCount: Int get() = documentRenderer.pageCount

    val infoMap: Map<String, Any> get() =
        mapOf(
            "id" to id,
            "pagesCount" to pagesCount
        )

    fun close() {
        documentRenderer.close()
        fileDescriptor.close()
    }

    /**
     * Open page by page number (not index!)
     */
    fun openPage(pageNumber: Int): PdfRenderer.Page = documentRenderer.openPage(pageNumber - 1)
}
